src/scripts/modules/media/body/body.coffee

```diff
@@ -152,6 +152,27 @@ define (require) ->
             # NOTE: `.toggleClass()` explicitly requires a `false` (not falsy) 2nd argument
             $el.toggleClass('ui-has-child-title', $title.length > 0)
 
+          # Create subtitles in each chapter with corresponding ids as their href
+          $totalSubtitle = $temp.find("[data-depth='1']")
+          $totalId = $totalSubtitle.children('[data-type="title"]')
+          idLength = $totalId.length
+          countIndex = 0
+          realHref = window.location.pathname
+          #Determine if this is a outer page view or not
+          $location = $(".table-of-contents [href$='#{realHref}'] > .title")
+          # Keep adding subtitles until all subtitles are added
+          if $location.children("li").length < idLength
+            while countIndex < idLength
```

Comment:
And later I changed the map function provided by you a little bit , and now it works.
`a=$temp.find('section[data-depth="1"]')
          $items = $.map(a, (node) ->
          # Get the section id so we can link to it
            id = $(node).attr('id')
            # Get a copy of the section title to add to the link
            $title = $(node).find('> [data-type="title"]').clone()
            # Build the list item and link
            $listItem = $("<li/>")
            $link = $("<a href='##{id}'/>")
            $link.append($title)
            $listItem.append($link)
            # Return the list item
            $listItem
         )`